tests/fake-buildkite-agent.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
```

Comment:
Can you add a brief comment at the top of this file explaining what it does? Something like "this Bash script serves as a mock for the Buildkite agent; during the test suite, we don't have the real Buildkite agent available, so we mock it by using this script; the script works by reading and writing key-value pairs to a temp store located at `/tmp/metadata_store`".